### ISSUE TITLE
Curious about the reasoning behind not always calling push in _transform

### ISSUE THREAD

**Issue by Gyran** [kevinGodell/pipe2jpeg#8]
Thanks for this package, solved my problem perfectly! 

I'm kind of new to working with streams and I'm curious about the reasoning behind not always calling `push` in the `_transform` method. I want to be able to use `p2j.on('data', ...)` as I would normally on a stream.
[The documentation says](https://nodejs.org/docs/latest-v12.x/api/stream.html#stream_transform_transform_chunk_encoding_callback):
> The transform.push() method may be called zero or more times to generate output from a single input chunk, depending on how much is to be output as a result of the chunk.

But when the [lib checks](https://github.com/kevinGodell/pipe2jpeg/blob/master/index.js#L54) if `pipesCount > 0` the `data` event is never called. 

I'd be happy to submit a PR if it's something you want. But I'm guessing I'm missing something.

**Comment by kevinGodell** [comment]
Good questions. Streams were very hard for me to figure out a while back when I was trying to make this. There were no great tutorials available and many gotchas had to be resolved. 

One of those issues was the _pushing of data_. If there was not something on the other end receiving and reading the data, then the pipe buffer would fill up and stall, thus, I have to make sure that a receiving pipe exists before pushing.

At the time when I made this, the recommended and more efficient usage of pipes was to read the pipe, not to listen to its _data_ event. I do include the _jpeg_ event as a convenience to those who don't prefer reading from the pipe.

If you were just to listen to the _data_ event on the pipe, then you would not always receive the complete jpeg file. This was an issue that I see often with piping. Many developers, such as myself at the time, assume that a complete file is always emitted on _data_. Depending on the OS, the buffer sizes are different. I think its ~8128 for mac, ~90000 for windows, and maybe ~65000 on linux. I ran into the problem that if the piped jpeg was bigger than the buffer size, the jpeg file would be incomplete. This would usually give the result of a jpeg that looks "torn in half". This limitation exists when piping data into nodejs from external software, such as ffmpeg. When piping between js libraries already running within nodejs, there does not seem to be this limitation. And for the life of me, I could not figure out a way to increase the buffer size for mac, as to prevent so many loops. I am sure the macOS developers would call that limitation a _feature_ instead of a _bug_.

Hopefully, I have given an answer that clarifies some things. Thanks.
